### PR TITLE
feat(celo): ubeswap integration in mask

### LINKED ISSUE #5824: [Bounty] Integrate Ubeswap, an AMM Protocol for Celo Assets into Mask Plugin
## Catalogue

- Platform
  - [x] Gitcoin
  - [ ] Hackerlink
  - [ ] Mask
- Experience
  - [ ] Beginner
  - [x] Intermediate
  - [ ] Advanced
- Category
  - [ ] Mask Core Feature
  - [x] Mask Plugin
  - [ ] Security/Cryptography Feature
  - [ ] Documentation
  - [ ] Design
  - [ ] Other
    - [ ] /_List it here_/
- Time Commitment
  - [x] 2 weeks
- Bounty Size
  - [x] 1000 USDC of $MASK equivalent 

## Part 1: Problem Description
**Background** Ubeswap is a decentralized exchange and automated market maker protocol for Celo assets.
Ubeswap is based on elements from Uniswap, Sushiswap, Synthetix, and Compound. This is to ensure that the Ubeswap ecosystem can support as many existing tools and projects from the Ethereum ecosystem as possible.

Ubeswap's goal is to increase liquidity on Celo. We envision a world where anyone can trade any asset from their phone or browser, without having to rely on decades-old clearinghouse technology or other centralized systems. This is only possible if people can invest in good assets. Ubeswap seeks to bring in liquidity from all of DeFi by harnessing the fast transaction times, high block gas limit, and stablecoin system of Celo.

**Solution**  Create a plugin that integrates Ubeswap into Mask. The UI should be identical to other swaps.

image

image

_Ubeswap Page_ 

Screen Shot 2022-03-07 at 5 42 05 PM

## Part 2: Resources

- [Bounty Guide](https://github.com/DimensionDev/Maskbook/tree/master/docs/bounty-guide.md)
- https://ubeswap.org/
- https://docs.ubeswap.org/

## Part 3: Contact and Support
@yisiliu @tzhan28 @LindaMask for ideas
@Jack-Works for implementations
Mask Network discord server: https://discord.gg/4Xccj7a75Q,
Ubeswap Discord: https://discord.com/invite/zZkUXCMPGP

Please speak up in the Mask discord's bounty section or just discord dm @yisiliu or @taylor zhang or @lin_dao #5824  so that we can find you on Discord. For faster communication reach out to lin_DAO on telegram and she will create a dedicated group with the Mask team to start the bounty process

Submission Requirements
Please include demo to show the integration is working.

Judging Criteria
Have a working demo.
Clean code & tech design
Clear UI, easy to use and generally good user experience

### PR DESCRIPTION
**Pull request by amritkumarj**
## Description
Ubeswap DEX(Uniswap Like) integration in Mask

Closes #5824

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Previews
photo6143144307209645749

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
  - [x] I have removed all in development `console.log`s
  - [x] I have removed all commented code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [x] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [x] I have delegated all web requests to the background service via the internal RPC bridge.

Full test video:

https://user-images.githubusercontent.com/72156537/157594785-6a1febdd-2e2b-4291-b5da-2968a648d28b.mp4

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

**Review by UncleBill — DISMISSED** [review]
LGTM. I like your screenshot and video.

**Comment by guanbinrui** [comment]
@DimensionDev/qa

**Review by UncleBill — DISMISSED** [review]

**Comment by UncleBill** [comment]
@DhruvJain1122 CI fails at eslint, syncing with the main branch (`develop`) should resolve it.

**Comment by wenluomask** [comment]
1. Please add Celo in Network tab; (Maybe just close temporarily)
Screen Shot 2022-03-20 at 12 02 00 AM

2.  Since Uberswap use approve button in app, we shouldn't use wrap button. It is supposed to be the unlock button;
3. The received amount is incorrect;
Screen Shot 2022-03-20 at 12 05 39 AM
Screen Shot 2022-03-20 at 12 05 50 AM

4. Didn't receive the estimate gas price;
Screen Shot 2022-03-20 at 12 09 20 AM

**Comment by amritkumarj** [comment]
@wenluomask 
1. Added Celo
image

2. Only in Celo Native Asset maybe due to Celo Native Asset address, same issue is in sushiswap finding what is causing this

3. Same as 2

4. Gas price showing correctly in my swap, try to swap anything other than native asset
image

File: packages/web3-constants/evm/trader.json [hunk 1]
```diff
@@ -1098,5 +1098,85 @@
...
+        "BSC": "",
+        "BSCT": "",
+        "Matic": "",
+        "Mumbai": "",
+        "Arbitrum": "",
+        "Arbitrum_Rinkeby": "",
+        "xDai": "",
+        "Avalanche": "",
+        "Avalanche_Fuji": "",
+        "Celo": "0xe3d8bd6aed4f159bc8000a9cd47cffdb95f96121",
```
Comment (UncleBill) [inline packages/web3-constants/evm/trader.json:1558]: I have double checked `0xe3d8bd6aed4f159bc8000a9cd47cffdb95f96121`

File: packages/web3-constants/evm/trader.json [hunk 2]
```diff
@@ -1098,5 +1098,85 @@
...
+        "BSC": "",
+        "BSCT": "",
+        "Matic": "",
+        "Mumbai": "",
+        "Arbitrum": "",
+        "Arbitrum_Rinkeby": "",
+        "xDai": "",
+        "Avalanche": "",
+        "Avalanche_Fuji": "",
+        "Celo": "0x62d5b84be28a183abb507e125b384122d2c25fae",
```
Comment (UncleBill) [inline packages/web3-constants/evm/trader.json:1581]: I have double checked `0x62d5b84be28a183abb507e125b384122d2c25fae`

File: packages/web3-constants/evm/trader.json [hunk 3]
```diff
@@ -1098,5 +1098,85 @@
...
+        "BSC": "",
+        "BSCT": "",
+        "Matic": "",
+        "Mumbai": "",
+        "Arbitrum": "",
+        "Arbitrum_Rinkeby": "",
+        "xDai": "",
+        "Avalanche": "",
+        "Avalanche_Fuji": "",
+        "Celo": "0xb3b8ff62960acea3a88039ebcf80699f15786f1b17cebd82802f7375827a339c",
```
Comment (UncleBill) [inline packages/web3-constants/evm/trader.json:1627]: I have double checked 0xb3b8ff62960acea3a88039ebcf80699f15786f1b17cebd82802f7375827a339c

**Review by UncleBill — DISMISSED** [review]

**Comment by guanbinrui** [comment]
> @wenluomask
> 
> 1. Added Celo
>    image
> 2. Only in Celo Native Asset maybe due to Celo Native Asset address, same issue is in sushiswap finding what is causing this
> 3. Same as 2
> 4. Gas price showing correctly in my swap, try to swap anything other than native asset
>    image

The native token of CELO is an ERC20 token. Is that mean we should add `<EthereumERC20TokenApprovedBoundary />` for the native token under the CELO network?

**Comment by wenluomask** [comment]
1 and 4 are fixed.

5. US dollar value is missing;
截屏2022-03-30 下午2 46 03

**Comment by guanbinrui** [comment]
Hi, @DhruvJain1122 Any updates?

**Comment by wenluomask** [comment]
Hello @DhruvJain1122 , could you merge latest develop into this pr and build again for me? I want to confirm that it is good to go except for #5.

File: packages/web3-constants/evm/token.json [hunk 4]
```diff
@@ -1210,7 +1210,7 @@
         "xDai": "0x0000000000000000000000000000000000000000",
         "Avalanche": "0x0000000000000000000000000000000000000000",
         "Avalanche_Fuji": "0x0000000000000000000000000000000000000000",
-        "Celo": "0x471ece3750da237f93b8e339c536989b8978a438",
+        "Celo": "0x0000000000000000000000000000000000000000",
```
Comment (UncleBill) [inline packages/web3-constants/evm/token.json]: Celo's native token is not '0x000' but like a ERC20 token.

**Review by UncleBill — APPROVED** [review]
LGTM. Please resolve the conflicts.

**Comment by wenluomask** [comment]
For the issue#2 that only happened in Celo Native Asset, you said maybe due to Celo Native Asset address. How was it going? @DhruvJain1122

**Comment by amritkumarj** [comment]
> For the issue#2 that only happened in Celo Native Asset, you said maybe due to Celo Native Asset address. How was it going? @DhruvJain1122

@wenluomask Not able to find the root cause of this issue

**Pull request closed without merging**